src/stim/diagram/detector_slice/detector_slice_set.cc

```diff
@@ -615,31 +619,37 @@ void DetectorSliceSet::write_svg_diagram_to(std::ostream &out) const {
 
     for (size_t k = 0; k < num_ticks; k++) {
         for (auto q : used_qubits()) {
-            auto c = coords(min_tick + k, q);
+            auto t = min_tick + k;
+            auto qc = coordsys.qubit_coords[q]; // the raw qubit coordinates
+            auto sc = coords(t, q); // the svg coordinates, offset to the correct slice plot
             out << "<circle";
-            write_key_val(out, "cx", c.xyz[0]);
-            write_key_val(out, "cy", c.xyz[1]);
+            write_key_val(out, "id", std::format("qubitdot_{}_{}_{}", qc.xyz[0], qc.xyz[1], t))
+            write_key_val(out, "cx", sc.xyz[0]);
+            write_key_val(out, "cy", sc.xyz[1]);
             write_key_val(out, "r", 2);
             write_key_val(out, "stroke", "none");
             write_key_val(out, "fill", "black");
             out << "/>\n";
         }
     }
 
-    // Boundaries around different slices.
+    // Border around different slices.
     if (num_ticks > 1) {
+        size_t k = 0;
         for (uint64_t col = 0; col < num_cols; col++) {
             for (uint64_t row = 0; row < num_rows && row * num_cols + col < num_ticks; row++) {
                 auto sw = coordsys.size.xyz[0];
                 auto sh = coordsys.size.xyz[1];
                 out << "<rect";
+                write_key_val(out, "id", std::format("border_{}_{}_{}", row, col, k))
                 write_key_val(out, "x", sw * col * SLICE_WINDOW_GAP);
                 write_key_val(out, "y", sh * row * SLICE_WINDOW_GAP);
                 write_key_val(out, "width", sw);
                 write_key_val(out, "height", sh);
                 write_key_val(out, "stroke", "black");
                 write_key_val(out, "fill", "none");
                 out << "/>\n";
+                k++;
```

Comment:
I think it would make sense to index using the absolute tick instead of this relative incrementing value of k.